cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -156,6 +158,11 @@ func (g *gitHubRepository) ComponentsPath() string {
 func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
 	release, err := g.getReleaseByTag(version)
 	if err != nil {
+		if errors.Is(err, errNotFound) {
+			// If it was ErrNotFound, then there is no release yet for the resolved tag.
+			// Ref: https://github.com/kubernetes-sigs/cluster-api/issues/7889
+			return nil, errors.Wrap(err, "release does not exist yet, consider setting \"GOPROXY=off\" as workaround")
```

Comment:
what about 
```suggestion
			return nil, errors.Wrap(err, "it looks like there is a new release being created in this moment, please retry later of set \"GOPROXY=off\" to get the current stable release")
```

Reply:
I am not so sure that in this error message we should say that "a new release is being created" if we get a `404 Not Found`.  Seems very specific.
How about: 
```go
return nil, errors.Wrapf(err, "release not found for version %s, please retry later or set \"GOPROXY=off\" to get the current stable release", version)
```